src/components/select/multi-select/multi-select.component.tsx

```diff
@@ -240,17 +240,20 @@ export const MultiSelect = React.forwardRef(
       textToMatch: string,
       list: React.ReactNode
     ) {
-      return (list as React.ReactElement[]).find((child) => {
-        const { text } = child.props;
+      return React.Children.toArray(list).find((child) => {
```

Comment:
**suggestion**: likewise what do you think of using a type predicate and `React.isValidElement` here, instead of the type assertions, so no potential type errors are masked?

Reply:
I can but as a side effect of this I'd probably need to also manipulate the type passed in the `handleTextboxChange` to something like:

```
        if (match) {
          setHighlightedValue(String(match.props.value));
        }
```